routing/heap.go

```diff
@@ -36,12 +36,15 @@ type nodeWithDist struct {
 	// Includes the routing fees and a virtual cost factor to account for
 	// time locks.
 	weight int64
+
+	// nextHop is the edge this route comes from.
+	nextHop *channeldb.ChannelEdgePolicy
 }
 
 // distanceHeap is a min-distance heap that's used within our path finding
 // algorithm to keep track of the "closest" node to our source node.
 type distanceHeap struct {
-	nodes []nodeWithDist
+	nodes []*nodeWithDist
```

Comment:
Now that this is holding a pointer, I think we should remember to nil the popped element.

Reply:
`Pop` is calling `delete` on this map already, if that's what you mean

Reply:
Should nil it to avoid gc leak, as here: https://golang.org/pkg/container/heap/#example__priorityQueue